js_modules/dagster-ui/packages/ui-core/src/assets/AssetsOverview.tsx

```diff
@@ -0,0 +1,335 @@
+
+type AssetCountsResult = {
+  countsByOwner: Record<string, number>;
+  countsByComputeKind: Record<string, number>;
+  countPerAssetGroup: CountPerGroupName[];
+  countPerCodeLocation: CountPerCodeLocation[];
+};
+
+type GroupMetadata = {
+  groupName: string;
+  repositoryLocationName: string;
+  repositoryName: string;
+};
+
+type CountPerGroupName = {
+  assetCount: number;
+  groupMetadata: GroupMetadata;
+};
+
+type CountPerCodeLocation = {
+  repoAddress: RepoAddress;
+  assetCount: number;
+};
+
+function buildAssetCountBySection(assets: AssetTableFragment[]): AssetCountsResult {
+  const assetCountByOwner: Record<string, number> = {};
+  const assetCountByComputeKind: Record<string, number> = {};
+  const assetCountByGroup: Record<string, number> = {};
+  const assetCountByCodeLocation: Record<string, number> = {};
+
+  assets.forEach((asset) => {
+    if (asset.definition) {
+      asset.definition.owners.forEach((owner) => {
+        const ownerKey = owner.__typename === 'UserAssetOwner' ? owner.email : owner.team;
+        assetCountByOwner[ownerKey] = (assetCountByOwner[ownerKey] || 0) + 1;
+      });
+
+      const computeKind = asset.definition.computeKind;
+      if (computeKind) {
+        assetCountByComputeKind[computeKind] = (assetCountByComputeKind[computeKind] || 0) + 1;
+      }
+
+      const groupName = asset.definition.groupName;
+      const locationName = asset.definition.repository.location.name;
+      const repositoryName = asset.definition.repository.name;
+
+      if (groupName) {
+        const metadata: GroupMetadata = {
+          groupName,
+          repositoryLocationName: locationName,
+          repositoryName,
+        };
+        const groupIdentifier = JSON.stringify(metadata);
+        assetCountByGroup[groupIdentifier] = (assetCountByGroup[groupIdentifier] || 0) + 1;
+      }
+
+      const stringifiedCodeLocation = JSON.stringify({
+        name: repositoryName,
+        location: locationName,
+      });
+      assetCountByCodeLocation[stringifiedCodeLocation] =
+        (assetCountByCodeLocation[stringifiedCodeLocation] || 0) + 1;
+    }
+  });
+
+  const countPerAssetGroup: CountPerGroupName[] = [];
+  Object.entries(assetCountByGroup).forEach(([groupIdentifier, count]) => {
+    const metadata: GroupMetadata = JSON.parse(groupIdentifier);
+    countPerAssetGroup.push({
+      assetCount: count,
+      groupMetadata: metadata,
+    });
+  });
+
+  return {
+    countsByOwner: assetCountByOwner,
+    countsByComputeKind: assetCountByComputeKind,
+    countPerAssetGroup,
+    countPerCodeLocation: Object.entries(assetCountByCodeLocation).reduce(
+      (a, [stringifiedRepoLocation, count]) =>
+        a.concat({repoAddress: JSON.parse(stringifiedRepoLocation), assetCount: count}),
+      [] as CountPerCodeLocation[],
+    ),
+  };
+}
+
+interface AssetOverviewCategoryProps {
+  children: React.ReactNode;
+  assetsCount: number;
+}
+
+function getGreeting() {
+  const hour = new Date().getHours();
```

Comment:
We'll want to make sure we takes the user's timezone preference into account.

Reply:
good call, added